program/rust/src/time_machine_types.rs

```diff
@@ -1,4 +1,36 @@
-#[derive(Debug, Clone)]
+use crate::c_oracle_header::{
+    pc_price_t,
+    EXTRA_PUBLISHER_SPACE,
+};
+use crate::error::OracleError;
+use bytemuck::{
+    Pod,
+    Zeroable,
+};
+use solana_program::msg;
+
+pub trait Tracker {
+    /// add the first price to a zero initialized tracker
+    fn add_first_price(
+        &mut self,
+        current_time: u64,
+        current_price: i64,
+        current_conf: u64,
+    ) -> Result<(), OracleError>;
+
+    ///add a new price to a tracker
+    fn add_price(
+        &mut self,
+        current_time: u64,
+        prev_time: u64,
+        current_price: i64,
+        prev_price: i64,
+        current_conf: u64,
+        prev_conf: u64,
```

Comment:
super nitpicky, but i think it would be better to sort the arguments like:

prev_time, 
prev_price
prev_conf,
current_time,
current_price,
current_conf,

this is likely easier for someone to remember because (1) previous comes before current, and (2) all the arguments for each are in a single block